liquibase-core/src/main/java/liquibase/lockservice/PostgresAdvisoryLockService.java

```diff
@@ -0,0 +1,165 @@
+package liquibase.lockservice;
+
+import liquibase.configuration.GlobalConfiguration;
+import liquibase.configuration.LiquibaseConfiguration;
+import liquibase.database.Database;
+import liquibase.database.core.PostgresDatabase;
+import liquibase.exception.DatabaseException;
+import liquibase.exception.LockException;
+import liquibase.executor.Executor;
+import liquibase.executor.ExecutorService;
+import liquibase.servicelocator.PrioritizedService;
+import liquibase.statement.core.RawSqlStatement;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.logging.Logger;
+
+public class PostgresAdvisoryLockService implements LockService {
+
+    private static final Logger LOGGER = Logger.getLogger(PostgresAdvisoryLockService.class.getName());
+
+    private static final int LOCK_NAMESPACE_ID = "liquibase".hashCode();
+    private static final int LOCK_ID = "lockservice".hashCode();
+
+    static final RawSqlStatement TRY_ACQUIRE_LOCK_SQL =
+        new RawSqlStatement(String.format("select pg_try_advisory_xact_lock(%d,%d)", LOCK_NAMESPACE_ID, LOCK_ID));
```

Comment:
Liquebase creates a lock per changeset, and a changeset could consist multiple changes.
Afaik since each change is run in it's own transaction, that would mean the lock would be released after the first change.

Reply:
thank you @gerbrand , I was not aware of one-tx per change. I had assumed that all changeset would be run in the same transaction. 

Can you point me to the entry point where all changeSet are applied? So that I can acquire one lock for all migrations, eg:

Reply:
If I read this correctly, Liquibase seems to be acquiring one lock per ChangeLog https://github.com/liquibase/liquibase/blob/master/liquibase-core/src/main/java/liquibase/Liquibase.java#L941 but I am not 100% sure on the tx scope because I don't see that being explicitly created anywhere.

Reply:
> Liquibase seems to be acquiring one lock per ChangeLog 

It acquires the global lock once for the changeset, which may contain multiple Changelogs.
The method you linked to, `StandardLockService.waitForLock()` will acquire the lock when first called, and return true if called a second or subsequent time.

Reply:
uhm, so I can probably change this implementation to use advisory session lock instead of tx locks. The lock will be bounded to the connection and it is automatically released when the connection is closed (so in the event of the VM crashing for instance).
What do you think ?

Reply:
Sounds good to me, yes.